synapse/datamodel.py

```diff
@@ -295,10 +292,16 @@ def getPropNorm(self, prop, valu, oldval=None):
 
         '''
         dtype = self.getPropType(prop)
-        if dtype == None:
-            return valu
+        if dtype != None:
+            try:
+                valu = dtype.norm(valu,oldval=oldval)
+            except BadTypeValu:
+                raise BadPropValu(name=prop, valu=valu)
+
+        if not s_compat.issystype(valu):
```

Comment:
It seems incongruous to me that getPropNorm could return a invalid system value.